scriptoocloud/src/test/java/edu/asu/diging/scriptoocloud/core/service/impl/DataFileServiceTest.java

```diff
@@ -0,0 +1,213 @@
+package edu.asu.diging.scriptoocloud.core.service.impl;
+
+import edu.asu.diging.scriptoocloud.core.data.DataFileRepository;
+import edu.asu.diging.scriptoocloud.core.data.DatasetRepository;
+import edu.asu.diging.scriptoocloud.core.exceptions.DataFileStorageException;
+import edu.asu.diging.scriptoocloud.core.exceptions.FileSystemStorageException;
+import edu.asu.diging.scriptoocloud.core.model.IDataFile;
+import edu.asu.diging.scriptoocloud.core.model.IDataset;
+import edu.asu.diging.scriptoocloud.core.model.impl.DataFile;
+import edu.asu.diging.scriptoocloud.core.model.impl.Dataset;
+import edu.asu.diging.simpleusers.core.model.IUser;
+import edu.asu.diging.simpleusers.core.model.impl.SimpleUser;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.*;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.support.AnnotationConfigContextLoader;
+
+import java.time.OffsetDateTime;
+import java.util.*;
+
+/**
+ * Test class for DataFileService
+ *
+ * @author John Coronite
+ */
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = AnnotationConfigContextLoader.class)
+class DataFileServiceTest {
+
+    @InjectMocks
+    private DataFileService dataFileService;
+
+    @Mock
+    private DataFileRepository dataFileRepository;
+
+    @Mock
+    private DatasetRepository datasetRepository;
+
+    @Mock
+    private FileSystemService fileSystemService;
+
+    private static final Long DATASET_ID = 1L;
+
+    private static final String DATASET_ID_STRING = "1";
+
+    private static final Long DATAFILE_ID = 2L;
+
+    private static final int PAGE_SIZE = 25;
+
+    private static final int SECOND_PAGE_SIZE = 10;
+
+    private static final int PAGE_NUMBER = 1;
+
+    private static final String USERNAME = "username";
+
+    private static final String DATASET_NAME = "datasetName";
+
+    private static final String FILE_NAME = "fileName.txt";
+
+    private static final String FILE_EXTENSION = "txt";
+
+    private static final String FILE_TYPE = Dataset.class.getSimpleName().toLowerCase(Locale.ROOT);
+
+    private static final String INDEX_BASED_FILENAME = "2.txt";
+
+    private static final byte[] BYTES = "abc".getBytes();
+
+    private static final OffsetDateTime CREATED_AT = OffsetDateTime.now();
```

Comment:
Out of curiosity, is this because there is only going to be one instance of each test class, so there is no benefit to static constants?  Are there any downsides to making constants static?  Again, just curious.

Reply:
There is probably no downside of it, its just not very consistent since all the other variables are non-static. And yes, there will be just the one instance so what's the point. Generally, I would avoid static members in classes unless they really serve a purpose. They either easily introduce bugs (if they are not final and someone forgets that it's a shared variable) or they kind of hard-code something that can't be changed programmatically later. E.g. you have a class that has a static final property `maxUsers` and you set that to say 20. If the requirements change and now it should be possible to dynamically change that number, you have to rewrite your code. Does that make sense?